src/PreCICE.jl

```diff
@@ -1332,5 +1340,149 @@ function actionReadIterationCheckpoint()
+end
+
+@doc """
+
+    writeScalarGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+
+Write gradient data of a scalar data, the value of a specified vertex to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndex::Integer`: Indice of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a scalar data for a 3D problem with 5 vertices:
+
+```julia
+data_id = 1
+vertex_id = 5
+gradientValues = [1.0, 2.0, 3.0]
+writeScalarGradientData(data_id, vertex_id, gradientValue)
+```
+
+"""
+function writeScalarGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+    ccall((:precicec_writeScalarGradientData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cdouble}), dataID, valueIndex, gradientValues)
+end
+
+
+@doc """
+
+    writeVectorGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+
+Write gradient data of a vector data, the value of a specified vertex to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndex::Integer`: Indice of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a vector data for a 3D problem with 5 vertices:
+
+```julia
+data_id = 1
+vertex_id = 5
+gradientValues = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 1.0, 2.0, 3.0]
+writeVectorGradientData(data_id, vertex_id, gradientValue)
+```
+"""
+function writeVectorGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+    ccall((:precicec_writeVectorGradientData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cdouble}), dataID, valueIndex, gradientValues)
+end
+
+@doc """
+
+writeBlockScalarGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+
+
+Write gradient data of a scalar data as a block, the value of a specified vertices to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndices::AbstractArray{Cint}`: Indices of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a vector data for a 2D problem with 3 vertices:
+
+```julia
+data_id = 1
+valueIndices = [1,2,3]
+gradientValues = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0]
+writeBlockScalarGradientData(data_id, vertex_id, gradientValue)
+```
+
+"""
+function writeBlockScalarGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+    _size = length(gradientValues)
```

Comment:
See above, because we have a vector for each datapoint, we should use a matrix or give the dimensions and format in which the data should be provided